cli.go

```diff
@@ -24,17 +32,57 @@ name and description will be used to construct the help message for the app:
 */
 func App(name, desc string) *Cli {
 	return &Cli{
-		&Cmd{
+		Cmd:&Cmd{
 			name:          name,
 			desc:          desc,
 			optionsIdx:    map[string]*opt{},
 			argsIdx:       map[string]*arg{},
 			ErrorHandling: flag.ExitOnError,
 		},
+		version: nil,
 	}
 }
 
 /*
+Version sets the version string of the CLI app together with the arguments that can be used to trigger
+printing the version string via the CLI.
+
+	Usage: appName --$name
+	$version
+
+ */
+func (cli *Cli) Version(name, version string) {
+	cli.BoolOpt(name, false, "Show the version and exit")
+	namesSl := strings.Split(name, " ")
+	option := cli.optionsIdx["-" + namesSl[0]]
```

Comment:
This won't correctly with the long options (which must be prefixed by `--`).
See the `mkOpt` function in `options.go` for how to handle these.
I would suggest extracting the `name -> []string` to a reusable function.